src/rename.rs

```diff
@@ -138,7 +155,9 @@ impl Rename for PatKind {
         match self {
             Self::Var(v) => {
                 if !r.top_scope().insert(*v) {
-                    todo!("duplicate variable in pattern, TODO: report error");
+                    r.push_diagnostic(
+                        Diagnostic::new(*v, format!("Duplicate viariable '{}' in pattern", v.text(r.db)))
```

Comment:
Hmm, diagnostics don't have source spans? Maybe we should rethink that before adding a bunch of them